libraries/@mattermost/secure-pdf-viewer/android/build.gradle

```diff
@@ -0,0 +1,69 @@
+buildscript {
+  ext.safeExtGet = {prop, fallback ->
+    rootProject.ext.has(prop) ? rootProject.ext.get(prop) : fallback
+  }
+  repositories {
+    google()
+    gradlePluginPortal()
+  }
+  def kotlin_version = rootProject.ext.has('kotlinVersion') ? rootProject.ext.get('kotlinVersion') : "1.9.25"
+  dependencies {
+    classpath("com.android.tools.build:gradle:8.6.0")
+    classpath("org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version")
+  }
+}
+
+def isNewArchitectureEnabled() {
+    return project.hasProperty("newArchEnabled") && project.newArchEnabled == "true"
+}
+
+apply plugin: 'com.android.library'
+apply plugin: 'org.jetbrains.kotlin.android'
+if (isNewArchitectureEnabled()) {
+    apply plugin: 'com.facebook.react'
+}
+
+android {
+  compileSdkVersion safeExtGet('compileSdkVersion', 34)
+  namespace "com.mattermost.securepdfviewer"
+
+  defaultConfig {
+      minSdkVersion safeExtGet('minSdkVersion', 24)
+      targetSdkVersion safeExtGet('targetSdkVersion', 34)
+      buildConfigField("boolean", "IS_NEW_ARCHITECTURE_ENABLED", isNewArchitectureEnabled().toString())
+  }
+
+  packagingOptions {
+        exclude 'lib/arm64-v8a/libc++_shared.so'
+        exclude 'lib/x86_64/libc++_shared.so'
+        exclude 'lib/armeabi-v7a/libc++_shared.so'
+        exclude 'lib/x86/libc++_shared.so'
+    }
+
+  sourceSets {
+    main {
+        if (isNewArchitectureEnabled()) {
+            java.srcDirs += ['src/newarch']
+        } else {
+            java.srcDirs += ['src/oldarch']
+        }
+    }
+}
+}
+
+repositories {
+  google()
+  mavenCentral()
+  maven { url 'https://jitpack.io' }
+  maven {
+      // All of React Native (JS, Obj-C sources, Android binaries) is installed from npm
+      url "$projectDir/../node_modules/react-native/android"
+  }
+}
+
+dependencies {
+    implementation 'com.facebook.react:react-native'
+    implementation('com.github.enahum:AndroidPdfViewer:3.2.0-pdfFile.6')
```

Comment:
@enahum I just started my review but what's the benefit of forking if we're up to date with the initial repo? Also why did we prefer forking from a fork rather than from the original library? Are there additions there that we benefit from?

Reply:
@esarafianou a iut the fork of the fork is cause the original one is too outdated while the fork from which we forked has newer versions of libraries.

Then we did fork and we are using another branch, not master, in it, we exposed a prop that was private in order to be able to detect certain gestures and act on them like when a link in the document is pressed.

Reply:
Pdfrenderer is only api 35+ so not really an option and pdf.js I could not make it work

The other libraries I found are either paid or too limited

Reply:
Wait that renderer seems available in api 21, hmm nope did not try that one I got confused with https://developer.android.com/jetpack/androidx/releases/pdf

Reply:
Looks like pdfRenderer does not support password protection on the pdf files

Reply:
If PdfRenderer looks otherwise suitable, could we use [qpdf](https://github.com/qpdf/qpdf) for decryption and then feed the decrypted file to the renderer?

Reply:
Ok is an option, I would need to add all the gesture system, a way to allow internal and if the setting is enabled external links and so on.. but worth taking a look and see if by the end of it we are able to match what we have here.

In the meantime please review all the rest?

Reply:
I had a very brief look at all of it and didn't spot anything else terribly concerning. I'm not in the office again until next Wednesday, but if there's still reviewing to do then, I'll pick up where I left off. In the meantime, @esarafianou can continue the security review if she finds the time; I know there's a lot going on.